reactor-core/src/test/java/reactor/util/concurrent/SpscArrayQueueTest.java

```diff
@@ -119,4 +125,44 @@ public void shouldNotRemoveElementOnPeek() {
 			assertThat(q.peek()).isEqualTo(1);
 			assertThat(q.size()).isEqualTo(2);		}
 	}
+
+	@Test
+	@Tag("slow")
+	void objectPadding() {
+		ClassLayout layout = ClassLayout.parseClass(SpscArrayQueue.class);
+
+		AtomicLong currentPaddingSize = new AtomicLong();
+		List<String> interestingFields = new ArrayList<>();
+		List<Long> paddingSizes = new ArrayList<>();
+
+		layout.fields().forEach(field -> {
+			if (field.name().startsWith("pad")) {
+				currentPaddingSize.addAndGet(field.size());
+			}
+			else {
+				if (currentPaddingSize.get() > 0) {
+					interestingFields.add("[padding]");
+					paddingSizes.add(currentPaddingSize.getAndSet(0));
+				}
+				interestingFields.add(field.name());
+			}
+		});
+		if (currentPaddingSize.get() > 0) {
+			interestingFields.add("[padding]");
+			paddingSizes.add(currentPaddingSize.getAndSet(0));
+		}
+
+		assertThat(interestingFields).containsExactly(
+				"array",
+				"mask",
+				"[padding]",
+				"producerIndex",
+				"[padding]",
+				"consumerIndex",
+				"[padding]"
+		);
+
+		final long minPaddingSize = 120;
+		assertThat(paddingSizes).filteredOn(padSize -> padSize >= minPaddingSize).hasSize(3);
```

Comment:
same as above, filtering is not really useful for the assertion here (and potentially subtly error prone), could be simply `assertThat(paddingSizes).containsExactly(128L, 128L, 128L);`

Reply:
Fixed